Source/JavaScriptCore/b3/air/AirCCallingConvention.cpp

```diff
@@ -37,14 +37,22 @@ namespace JSC { namespace B3 { namespace Air {
 namespace {
 
 template<typename BankInfo>
-Arg marshallCCallArgumentImpl(unsigned& argumentCount, unsigned& stackOffset, Value* child)
+void marshallCCallArgumentImpl(Vector<Arg>& result, unsigned& argumentCount, unsigned& stackOffset, Value* child)
 {
-    unsigned argumentIndex = argumentCount++;
-    if (argumentIndex < BankInfo::numberOfArgumentRegisters)
-        return Tmp(BankInfo::toArgumentRegister(argumentIndex));
+    if (is32Bit() && child->type() == Int64)
+        argumentCount = WTF::roundUpToMultipleOf<2>(argumentCount);
```

Comment:
So I guess there are an even number of argument registers on ARM?

Reply:
There happen to be, yes, but also this is a requirement of the ABI:

>  C.3 | If the argument requires double-word alignment (8-byte), the NCRN is rounded up to the next even register number.

https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst#parameter-passing